internal/controller/rhdh/backstage.go

```diff
@@ -36,121 +34,161 @@ var ConfigMapNameAndConfigDataKey = map[string]string{
+			existingSubscription.Spec = rhdhSubscription.Spec
+			if err := client.Update(ctx, existingSubscription); err != nil {
+				rhdhLogger.Error(err, "Error occurred when updating subscription spec", "SubscriptionName", BackstageSubscriptionName)
+				return err
+			}
+			rhdhLogger.Info("Successfully updated subscription spec", "SubscriptionName", BackstageSubscriptionName)
+		}
+	}
+	return nil
+}
+
+func CreateRHDHSecret(secretNamespace string, ctx context.Context, client client.Client) error {
 	logger := log.FromContext(ctx)
 	logger.Info("Creating Backstage NPMrc Secret")
 
 	secret := &corev1.Secret{}
 	err := client.Get(ctx, types.NamespacedName{
 		Namespace: secretNamespace,
-		Name:      secretName,
+		Name:      RegistrySecretName,
 	}, secret)
 
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			logger.Info("Secret does not exist. Creating secret", "Secret", secretName)
+			logger.Info("Secret does not exist. Creating secret", "Secret", RegistrySecretName)
 			newSecret := &corev1.Secret{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      secretName,
+					Name:      RegistrySecretName,
 					Namespace: secretNamespace,
-					Labels:    operations.AddLabel(),
+					Labels:    kubeoperations.AddLabel(),
 				},
 				Type: corev1.SecretTypeOpaque,
 				StringData: map[string]string{
-					".npmrc": fmt.Sprintf("registry=%s", npmRegistry),
+					".npmrc": fmt.Sprintf("registry=%s", NpmRegistry),
 				},
 			}
 
 			if err := client.Create(ctx, newSecret); err != nil {
-				logger.Error(err, "Error occurred when creating secret", "Secret", secretName)
+				logger.Error(err, "Error occurred when creating secret", "Secret", RegistrySecretName)
 				return err
 			}
-			logger.Info("Successfully created secret", "Secret", secretName)
+			logger.Info("Successfully created secret", "Secret", RegistrySecretName)
 			return nil
 		}
-		logger.Error(err, "Error occurred when checking secret exist", "Secret", secretName)
+		logger.Error(err, "Error occurred when checking secret exist", "Secret", RegistrySecretName)
 		return err
 	}
-	logger.Info("Secret already exist", "Secret", secretName)
+	logger.Info("Secret already exist", "Secret", RegistrySecretName)
 	return nil
 }
 
-func HandleCRCreation(
-	operator orchestratorv1alpha1.RHDHOperator,
-	pluginsDetails orchestratorv1alpha1.RHDHPlugins,
-	clusterDomain string,
+func HandleRHDHCR(
+	rhdhConfig orchestratorv1alpha1.RHDHConfig,
+	argoCDEnabled, tektonEnabled bool,
+	clusterDomain, wfNamespace string,
 	ctx context.Context, client client.Client) error {
-	bsLogger := log.FromContext(ctx)
+	rhdhLogger := log.FromContext(ctx)
 
-	bsLogger.Info("Handling Backstage resources")
+	rhdhLogger.Info("Handling Backstage resources")
 
-	bsConfigMapList := GetConfigmapList(ctx, client, clusterDomain, operator, pluginsDetails)
+	bsConfigMapList := GetConfigmapList(ctx, client, clusterDomain, wfNamespace, argoCDEnabled, tektonEnabled, rhdhConfig)
```

Comment:
I think you renamed `wfNamespace` and other `wf` variables elsewhere, you may have missed this one (and some others)